common/primitives/core/src/intent.rs

```diff
@@ -47,3 +51,27 @@ pub struct TransferSolana {
     pub to: [u8; 32],
     pub value: u64,
 }
+
+#[derive(Encode, Decode, Debug, Clone, PartialEq, Eq, MaxEncodedLen, TypeInfo)]
+pub struct Asset {
+    pub id: AssetId,
+    pub chain_id: ChainId,
+}
+
+#[derive(
+    Encode, Decode, Copy, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen, Ord, PartialOrd, Debug,
+)]
+pub enum MultiAddress {
+    Address32(Address32),
+    Address20(Address20),
+    Address33(Address33),
+}
+
+#[derive(Encode, Decode, Debug, Clone, PartialEq, Eq, MaxEncodedLen, TypeInfo)]
+pub struct SwapOrder {
+    pub from_asset: Asset,
+    pub from_amount: u64,
+    pub to_asset: Asset,
+    pub to_min_amount: u64,
+    pub to_address: Option<MultiAddress>,
```

Comment:
We might need slippage tolerance here. Can be added later.

Reply:
The `to_min_amount` should be calculated with the slippage and the quote from the CEX/DEX we'll use. (the min amount of the output token is what the raydium sdk uses for Swap Instructions, it took it as a guide for now)